File: web-development/level-1/helsinki-part-1-2/Java/part1/week5/week05-ex85-Lyrra/CashRegister.java
public class CashRegister {
  private double cashInRegister; //amount of cash in the register
  private int economicalSold; //amount of economical lunches sold
  private int gourmetSold; //amount of gourmet linches sold

  //constructor
  public CashRegister() {
    this.cashInRegister = 1000;
  }

  public double payEconomical(double cashGiven) {
    double economicPrice = 2.50;
    if (cashGiven >= economicPrice)  {
      this.cashInRegister += economicPrice;
      this.economicalSold++;
      cashGiven -= economicPrice;
      return cashGiven;
    }
    else  {
      System.out.print("The money is not enough :");
      return cashGiven;
    }
  }

  public double payGourmet(double cashGiven)  {
    double gourmetPrice = 4.00;
    if (cashGiven >= gourmetPrice)  {
      this.cashInRegister += gourmetPrice;
      this.gourmetSold++;
      cashGiven -= gourmetPrice;
      return cashGiven;
    }
    else  {
      System.out.print("The money is not enough: ");
      return cashGiven;
    }
  }


  public String toString()  {
    return "money in register "+this.cashInRegister+" economical lunches sold:"+this.economicalSold+" gourmet lunches sold: "+this.gourmetSold;
  }

  

//class ends heres
}

File: web-development/helsinki-moocs/python/week05/ex86/CashRegister.py
class CashRegister:

    #constructor
    def __init__(self):
        self.cash_in_register = 1000.00
        self.economical_sold = 0
        self.gourmet_sold = 0

    #behaviors

    #pays economically 2.50 $
    def pay_economical(self, cash_given):
        if cash_given >= 2.50:
            self.cash_in_register += 2.50
            self.economical_sold += 1
            return 'Cash: $', cash_given - 2.50
        else:
            return 'Cash given is not enough : $',cash_given

    #pays gourmet 4.00 $
    def pay_gourmet(self, cash_given):
        if cash_given >= 4.00:
            self.cash_in_register += 4.00
            self.gourmet_sold += 1
            return 'Cash: $', cash_given - 4.00
        else:
            return 'Cash given is note enough: $',cash_given

    #returns a toString method of the register
    def register_info(self):
        return 'Money in register: $', self.cash_in_register, '\
        economical sold: ', self.economical_sold, '\
        gourmet sold: ', self.gourmet_sold
